### PR TITLE
[IA-5008] Update dataproc version + hail image

### PR DESCRIPTION
**Pull request by lucymcnatt**
__Jira ticket__: https://broadworkbench.atlassian.net/browse/[ticket_number]

### What

-

### Why

-

### What to test

- 

### Who tested and where

- [ ] This change is covered by automated tests 
  - _NB: Rerun automation tests on this PR by commenting `jenkins retest` or `jenkins multi-test`._
- [ ] I validated this change 
- [ ] Primary reviewer validated this change 
- [ ] I validated this change in the dev environment

### CODE REVIEW

File: jenkins/dataproc-custom-images/create_dataproc_image.sh [hunk 1]
```diff
@@ -36,7 +36,7 @@ gcloud config set dataproc/region us-central1
 
 python generate_custom_image.py \
     --image-name "$OUTPUT_IMAGE_NAME" \
-    --dataproc-version "2.1.11-debian11" \
+    --dataproc-version "2.2.5-debian12" \
```
Comment (rtitle) [inline jenkins/dataproc-custom-images/create_dataproc_image.sh]: Looks like the latest is `2.2.21-debian12`, not sure if we want that?

https://cloud.google.com/dataproc/docs/concepts/versioning/dataproc-release-2.2

**Review by rtitle — COMMENTED** [review]
We probably also want a terra-docker PR, and update the terra-jupyter-hail image version in `prepare-custom-leonardo-jupyter-dataproc-image.sh`, right?

Edit: I made https://github.com/DataBiosphere/terra-docker/pull/489 quickly

File: jenkins/dataproc-custom-images/prepare-custom-leonardo-jupyter-dataproc-image.sh [hunk 2]
```diff
@@ -145,12 +145,16 @@ dpkg --configure -a
 # This line fails consistently, but it does not fail in a fatal way so we add `|| true` to prevent the script from halting execution
 # The message that is non-fatal is `Sub-process /usr/bin/dpkg returned an error code (1).`
 # NOTE: If it fails with another legitimate error, this `|| true` could mask it. It was used as a last resort after a lot of attempts to fix.
-apt-get install -y -q docker-ce || true
+# apt-get install -y -q docker-ce || true
+
+# start docker
+systemctl start docker
```
Comment (rtitle) [inline jenkins/dataproc-custom-images/prepare-custom-leonardo-jupyter-dataproc-image.sh:151]: This looks promising

**Review by cindy-broadinstitute — APPROVED** [review]

**Comment by rtitle** [comment]
Looks like the [job passed](https://github.com/DataBiosphere/leonardo/actions/runs/9842766234). 🎉 

I guess the next step would be to update the `customDataprocImage` in `reference.conf`, and test.

**Review by rtitle — APPROVED** [review]

**Pull request closed without merging**